e2e/process_utils_test.go

```diff
@@ -0,0 +1,227 @@
+package e2e
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	toml "github.com/pelletier/go-toml"
+)
+
+type testProcess struct {
+	Cmd *exec.Cmd
+}
+
+// IsRunning will return true if the process is currently
+// running.
+func (tp *testProcess) IsRunning() bool {
+	process, err := os.FindProcess(tp.Cmd.Process.Pid)
+	if err != nil {
+		return false
+	}
+
+	if err := process.Signal(syscall.Signal(0)); err != nil {
+		return false
+	}
+
+	return true
+}
+
+// Stop will terminate the associated process. It will attempt a graceful
+// shutdown before killing the process.
+func (tp *testProcess) Stop() error {
+	tp.Cmd.Process.Signal(os.Interrupt) // try shutting down gracefully
+	time.Sleep(500 * time.Millisecond)
+	if tp.IsRunning() {
```

Comment:
suggestion:

can we simplify this

```
if tp.IsRunning() {
	time.Sleep(1 * time.Second)
        return tp.Cmd.Process.Kill()
        } 
return nil
	
	
```

Reply:
I don't see how that's simpler. The current code:
* Signals the process to shut down
* Waits a 1/2 second
* Checks to see if process is still alive
* If it is, tries to kill the process

It seems simple enough to me, and it pretty much has to wait before running the IsRunning test, not after.

Reply:
I just wanted to remove else case, it won't affect functionality

Reply:
Oh! I understand you. Your example moved the sleep and would have resulted in a behavior change, but I get you now. Something for another day. :-)